src/Chainweb/Pact/PactService.hs

```diff
@@ -162,9 +164,30 @@ withPactService ver cid chainwebLogger bhDb pdb sqlenv config act =
                     , _psBlockGasLimit = _pactBlockGasLimit config
                     , _psChainId = cid
                     }
-            !pst = PactServiceState Nothing mempty initialParentHeader P.noSPVSupport
-        runPactServiceM pst pse $ do
+        let !pst = PactServiceState Nothing mempty initialParentHeader P.noSPVSupport
+
+        when (_pactRosettaEnabled config) $ do
```

Comment:
Do we really need the configuration flag for this? For efforts/configuration from the user's POV.

We could run the check before starting the rosetta HTTPServer, turning it off if there is no required data in the db for example.

Reply:
The node should exit if the full history is required for one of its features. However @edmundnoble had the idea that instead of this being whether rosetta is enabled or not, we have a more general flag like `_pactFullHistoryRequired :: Bool`.

Reply:
We can use this later too, to determine whether to enable incremental compaction. Even in the short term if we stop writing pending rows from each tx and just write the final rows in a block, we can use this to determine whether to do that.